dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotReactiveElasticsearchClient.java

```diff
@@ -0,0 +1,874 @@
+     */
+    @Override
+    public Mono<Void> createIndex(HttpHeaders headers, CreateIndexRequest createIndexRequest) {
+
+        return sendRequest(createIndexRequest, DotReactiveElasticsearchClient.RequestCreator.indexCreate(), AcknowledgedResponse.class, headers) //
+                .then();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.Indices#openIndex(org.springframework.http.HttpHeaders, org.elasticsearch.action.admin.indices.open.OpenIndexRequest)
+     */
+    @Override
+    public Mono<Void> openIndex(HttpHeaders headers, OpenIndexRequest request) {
+
+        return sendRequest(request, DotReactiveElasticsearchClient.RequestCreator.indexOpen(), AcknowledgedResponse.class, headers) //
+                .then();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.Indices#closeIndex(org.springframework.http.HttpHeaders, org.elasticsearch.action.admin.indices.close.CloseIndexRequest)
+     */
+    @Override
+    public Mono<Void> closeIndex(HttpHeaders headers, CloseIndexRequest closeIndexRequest) {
+
+        return sendRequest(closeIndexRequest, DotReactiveElasticsearchClient.RequestCreator.indexClose(), AcknowledgedResponse.class, headers) //
+                .then();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.Indices#refreshIndex(org.springframework.http.HttpHeaders, org.elasticsearch.action.admin.indices.refresh.RefreshRequest)
+     */
+    @Override
+    public Mono<Void> refreshIndex(HttpHeaders headers, RefreshRequest refreshRequest) {
+
+        return sendRequest(refreshRequest, DotReactiveElasticsearchClient.RequestCreator.indexRefresh(), RefreshResponse.class, headers) //
+                .then();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.Indices#updateMapping(org.springframework.http.HttpHeaders, org.elasticsearch.action.admin.indices.mapping.put.PutMappingRequest)
+     */
+    @Override
+    public Mono<Void> updateMapping(HttpHeaders headers, PutMappingRequest putMappingRequest) {
+
+        return sendRequest(putMappingRequest, DotReactiveElasticsearchClient.RequestCreator.putMapping(), AcknowledgedResponse.class, headers) //
+                .then();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.Indices#flushIndex(org.springframework.http.HttpHeaders, org.elasticsearch.action.admin.indices.flush.FlushRequest)
+     */
+    @Override
+    public Mono<Void> flushIndex(HttpHeaders headers, FlushRequest flushRequest) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'headers' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5113278)